### PR TITLE
Refactor ColdFront settings

### PR DESCRIPTION
**Pull request by aebruno**
Refactor ColdFront settings.

This PR attempts to better organize ColdFront settings and configuration. As the number of plugins grows having a well defined way to configure the ColdFront application will be imperative. The main goals are:

1. Split the config/settings across multiple files for better organization.
2. Allow configuring ColdFront from environment variables or a env file
3. Enable easy way to set custom config values system wide for production installs as well as dev/staging environments

In this commit we use django-environ to fetch config settings from the environment. We also use django-split-settings which provides a way to split up django settings across multiple files.

The organization is as follows:

- coldfront/config/base.py = Base Django config settings
- coldfront/config/auth.py = Authentication backends
- coldfront/config/database.py = Database backends
- coldfront/config/email.py = Email settings
- coldfront/config/logging.py = Logging settings
- coldfront/config/core.py = Core ColdFront settings
- coldfront/config/plugins/* = ColdFront Plugins

Sites can override any config settings by creating a local_settings.py file in coldfront/config, or /etc/coldfront/local_settings.py.  Additionally, they can override settings using environment variables, or an environment file. ColdFront will search for environment files in this order: .env (in coldfront project directory) or /etc/coldfront/coldfront.env. You can also specify the path to the environment file using COLDFRONT_ENV.

### CODE REVIEW

File: coldfront/plugins/freeipa/README.md [hunk 1]
```diff
@@ -62,30 +62,21 @@ $ sss_cache -E
```
Comment (dsajdak) [inline coldfront/plugins/freeipa/README.md:62]: Under Design section:  FreeIPA unix groups can be set on a per allocation basis using using a allocation attribute named "freeipa_group" - remove double 'using' and change to 'using an allocation'

```suggestion
add 'active' to allocation info as these tasks only happen to active allocations.  Might want to also explain users are removed from freeipa group(s) when allocation expires.
```

File: docs/pages/config.md [hunk 2]
```diff
@@ -0,0 +1,237 @@
+# Configuration
+
+ColdFront can be configured via environment variables, an environment file, or
+a python file which can be used for more advanced configuration settings. This
+document covers the configuration settings available in ColdFront and the core
+plugins. For information on installing ColdFront [see here](install.md).
+
+## Configuration files
+
+You can set environment varialbes via a file and ColdFront by default will look
```
Comment (dsajdak) [inline docs/pages/config.md:10]: variables

File: docs/pages/config.md [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+## Configuration Variables
+
+### Base settings
+
+The following settings allow overriding basic ColdFront Django settings. For
+more advanced configuration use `local_settings.py`.
+
+| Name                 | Description                          |
+| :--------------------|:-------------------------------------|
+| ALLOWED_HOSTS        | A list of strings representing the host/domain names that ColdFront can server. [See here](https://docs.djangoproject.com/en/3.1/ref/settings/#allowed-hosts) |
```
Comment (dsajdak) [inline docs/pages/config.md:62]: serve

File: docs/pages/config.md [hunk 4]
```diff
@@ -0,0 +1,237 @@
...
+| DEBUG                | Turn on/off debug mode. Never deploy a site into production with DEBUG turned on. [See here](https://docs.djangoproject.com/en/3.1/ref/settings/#debug) |
+| SECRET_KEY           | This is used to provide cryptographic signing, and should be set to a unique, unpredictable value. [See here](https://docs.djangoproject.com/en/3.1/ref/settings/#secret-key). If you don't provide this one will be generated each time ColdFront starts. |
+| LANGUAGE_CODE        | A string representing the language code. [See here](https://docs.djangoproject.com/en/3.1/ref/settings/#language-code1)
```
Comment (dsajdak) [inline docs/pages/config.md:65]: https://docs.djangoproject.com/en/3.1/ref/settings/#language-code

File: docs/pages/config.md [hunk 5]
```diff
@@ -0,0 +1,237 @@
...
+| EMAIL_SENDER                    | Default sender email address              |
+| EMAIL_SUBJECT_PREFIX            | Prefix to add to subject line             |
+| EMAIL_ADMIN_LIST                | List of admin email addresses.            |
+| EMAIL_TICKET_SYSTEM_ADDRESS     | Email address of ticketing system         |
+| EMAIL_DIRECTOR_EMAIL_ADDRESS    | Email address for director                | 
+| EMAIL_PROJECT_REVIEW_CONTACT    | Email address of review contact           |
+| EMAIL_DEVELOPMENT_EMAIL_LIST    | List of emails to send when in debug mode |
+| EMAIL_OPT_OUT_INSTRUCTION_URL   | URL of article regarding opt out          |
+| EMAIL_SIGNATURE                 | Email signature to add to outgoing emails |
+| EMAIL_ALLOCATION_EXPIRING_NOTIFICATION_DAYS   | List of days to send email notifications for expiring allocatioins. Default 7,14,30 |
```
Comment (dsajdak) [inline docs/pages/config.md:126]: allocations

File: docs/pages/config.md [hunk 6]
```diff
@@ -0,0 +1,237 @@
...
+| :------------------------------|:-------------------------------------|
+| PLUGIN_AUTH_OIDC               | Enable OpenID Connect Authentication Backend. Default False |
+| OIDC_OP_JWKS_ENDPOINT          | URL of JWKS endpoint                 |
+| OIDC_RP_SIGN_ALGO              | Signature algorithm                  |
+| OIDC_RP_CLIENT_ID              | Client ID                            |
+| OIDC_RP_CLIENT_SECRET          | Client secret                        |
+| OIDC_OP_AUTHORIZATION_ENDPOINT | OAuth2 authorization endpoint        |
+| OIDC_OP_TOKEN_ENDPOINT         | OAuth2 token endpoint                |
+| OIDC_OP_USER_ENDPOINT          | OAuth2 userinfo endpoint             |
+| OIDC_VERIFY_SSL                | Verify ssl Deafult True              |
```
Comment (dsajdak) [inline docs/pages/config.md:165]: default

File: docs/pages/install.md [hunk 7]
```diff
@@ -0,0 +1,140 @@
...
+- You can log in as `admin` with password `test1234`.
+- You can log in as a PI using username `ccollins` with password `test1234`.
+- You can log in as center director using username `michardson` with password `test1234`.
+- Password for all users is also `test1234`.
+
+!!! danger "Danger"
+    Do not run with this data loaded in production environment. The default
+    user accounts are created with weak passwords and should only be used for
+    testing purposes. If you do decide to use the test data, delete the user
+    accounts created and minimum.
```
Comment (dsajdak) [inline docs/pages/install.md:140]: "at minimum"

**Review by dsajdak — CHANGES_REQUESTED** [review]
Looks great except these couple of typos.  Tested in staging all all functionality appears to work as before.

File: docs/pages/config.md (hunk 2 above)
Comment (aebruno) [reply docs/pages/config.md]: Fixed

File: docs/pages/config.md (hunk 3 above)
Comment (aebruno) [reply docs/pages/config.md]: Fixed

File: docs/pages/config.md (hunk 4 above)
Comment (aebruno) [reply docs/pages/config.md]: Fixed

File: docs/pages/config.md (hunk 5 above)
Comment (aebruno) [reply docs/pages/config.md]: Fixed

File: docs/pages/config.md (hunk 6 above)
Comment (aebruno) [reply docs/pages/config.md]: Fixed

File: docs/pages/install.md (hunk 7 above)
Comment (aebruno) [reply docs/pages/install.md]: Fixed

**Review by dsajdak — APPROVED** [review]

**Pull request merged**